### PR TITLE
Fix lesson incrementing

### PR DESCRIPTION
**Pull request by awjuliani**
### Proposed change(s)

Fix issue with curriculum attempting to increment lesson even when there are no more lessons. 

[Todo] I haven't added an additional test for this. I might need some help from someone more familiar with the curriculum codebase.

### Types of change(s)

- [X] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by chriselion** [comment]
Can you add a unit test that would have caught this before (or if you add the validation to completion_criteria, test that it fails)?

You might need to merge master to get the yamato tests passing again. Let me know if they're still failing after that and I'll take a look.

**Comment by awjuliani** [comment]
@vincentpierre Makes sense. The question is then how to handle this. My preference would be to just throw a warning and ignore the completion criteria of the last lesson, rather than throwing an error.

**Comment by vincentpierre** [comment]
> My preference would be to just throw a warning and ignore the completion criteria of the last lesson, rather than throwing an error.

I agree.

**Comment by awjuliani** [comment]
Added a warning and a test to catch the user error originally reported.

File: ml-agents/mlagents/trainers/tests/test_env_param_manager.py [hunk 1]
```diff
@@ -154,13 +154,55 @@ def test_curriculum_conversion():
...
             yaml.safe_load(test_bad_curriculum_no_competion_criteria_config_yaml)
         )
 
 
+def test_curriculum_raises_all_completion_criteria_conversion():
+    with pytest.warns(TrainerConfigWarning):
+        RunOptions.from_dict(
+            yaml.safe_load(test_bad_curriculum_all_competion_criteria_config_yaml)
+        )
+
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/tests/test_env_param_manager.py:204]: Not a must do for this PR, but to make sure your change worked, can you make a bunch of calls to `param_manager.update_lessons` to make sure it does not raise an error. (the error you fixed line 134 of environment_parameter_manager.py)

Comment (awjuliani) [reply ml-agents/mlagents/trainers/tests/test_env_param_manager.py:204]: Added an explicit test for this.

**Pull request merged**